zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupConfig.java

```diff
@@ -84,6 +91,12 @@ public BackupConfig(Builder builder) {
     this.storageConfig = builder.storageConfig.orElse(null);
     this.backupStoragePath = builder.backupStoragePath.orElse("");
     this.namespace = builder.namespace.orElse("");
+    this.timetableEnabled = builder.timetableEnabled.orElse(false);
+    // If timetable storage path is not given, use the backup storage path
+    this.timetableStoragePath = builder.timetableStoragePath.orElse(backupStoragePath);
+    // If timetable backup interval is not given, use the regular backup interval
+    this.timetableBackupIntervalInMs =
+        builder.timetableBackupIntervalInMs.orElse(backupIntervalInMinutes * 60000L);
```

Comment:
NIT: I would suggest to use TimeUnit object to do the conversion instead of using this number times up. TimeUnit.MINUTES.toMillis(backupIntervalInMinutes)

Reply:
Good questions/suggestions. The 15 minute interval is sufficient for the timetable as a backup value. Normally, it is expected that the user populates this field.